snowdrop-core/src/Snowdrop/Core/BaseM.hs

```diff
@@ -82,15 +82,15 @@ instance Applicative (BaseM e eff ctx) where
     BaseM a <*> BaseM b = BaseM $ a <*> b
 
 instance Monad (BaseM e eff ctx) where
-    a >>= b = BaseM $ unBaseM a >>= unBaseM . b
+    a >>= b = unBaseM a >>= b
```

Comment:
As I understand, `unBaseM a` will use `>>=` of `BaseM` (as it operates in `forall m. Monad m`) and thus this method will perform a recursive indefinite loop

Reply:
Yeah, stuck into this recently.